mailpoet/assets/js/src/global.d.ts

```diff
@@ -39,13 +39,13 @@ interface JQuery {
 declare module '@woocommerce/blocks-checkout' {
   type CheckboxControlProps = {
     className?: string;
-    label?: string;
+    label?: string | React.ReactNode;
     id?: string;
-    instanceId?: string;
-    onChange?: (value: boolean) => void;
-    children?: React.ReactChildren | React.ReactElement;
+    onChange: (value: boolean) => void;
+    children?: React.ReactChildren;
     hasError?: boolean;
     checked?: boolean;
+    disabled?: string | boolean | undefined;
```

Comment:
I'm kind of new to the code. Is  a good practice to use "undefined" as a type? Would there be an option to initialize the disabled property in the code instead?

Reply:
This is an imported component, so I only copied types from its source code at https://github.com/woocommerce/woocommerce/blob/644c13b828e30a0fc0c7894b1960f8f38094299c/plugins/woocommerce-blocks/packages/components/checkbox-control/index.tsx#L23 - I don't know the reasoning to use `undefined` but from maintenance point of view, it's easier to use the same types, rather than trying to remove `undefined` and handle it on our side.